### ISSUE TITLE
[New Monitor] RTSP support

### ISSUE THREAD

**Issue by myxor** [louislam/uptime-kuma#2851]
### ⚠️ Please verify that this feature request has NOT been suggested before.

- [X] I checked and didn't find similar feature request

### 🏷️ Feature Request Type

New Monitor

### 🔖 Feature description

Support RTSP (https://en.wikipedia.org/wiki/Real_Time_Streaming_Protocol) which is often used by IP Cameras.

### ✔️ Solution

Support RTSP Monitor

### ❓ Alternatives

-

### 📝 Additional Context

-

**Comment by totalitarian** [comment]
love this too

**Comment by CommanderStorm** [comment]
As always: We are open to PRs, here is our [contribution guide](https://github.com/louislam/uptime-kuma/blob/master/CONTRIBUTING.md).

Note that
- a Pull request should be class extending `MonitorType` instead of the currently primary approach of writing monitors in `monitor.js`. See for details.
- Depending on how easy this is to test, a testcase is also appreciated.
  Rationale: I don't expect plenty of people to use this
  => If this breaks accidentally, we wouldn't notice for a long time

**Comment by Baruxa** [comment]
I need it very much, that would be great!!!

**Comment by serveradmin9912** [comment]
Yes I would need that too. Im currently just pinging the cams. A RTSP checkup would be more precise.

**Comment by thielj** [comment]
@CommanderStorm 

RTSP(S) is almost exactly like HTTP(S), except that you would send something like this...

```
OPTIONS rtsp://media.example.com:554/path?query RTSP/1.0
CSeq: 1
```

...and receive something like this...

```
RTSP/1.0 200 OK
Cseq: 1
```

...plus some optional headers you can ignore.
